src/fmeval/eval_algorithms/faithfulness.py

```diff
@@ -0,0 +1,297 @@
+Explanation: The context mentions the courses John is currently enrolled in, and Artificial Intelligence is not mentioned. Therefore, it cannot be deduced that John is taking a course on AI. Verdict: No.
+3. John is a dedicated student.
+Explanation: The prompt states that he spends a significant amount of time studying and completing assignments. Additionally, it mentions that he often stays late in the library to work on his projects, which implies dedication. Verdict: Yes.
+4. John has a part-time job.
+Explanation: There is no information given in the context about John having a part-time job. Therefore, it cannot be deduced that John has a part-time job.  Verdict: No.
+5. John is interested in computer programming.
+Explanation: The context states that John is pursuing a degree in Computer Science, which implies an interest in computer programming. Verdict: Yes.
+Final verdict for each statement in order: No. No. Yes. No. Yes.
+context:\n$context
+statements:\n$statements
+Answer: Assistant:
+"""  # noqa: E501
+
+
+class FaithfulnessScore(Transform):
+    """This transform augments its input record with the computed faithfulness score.
+
+    See the docstring for `Faithfulness` for more details regarding the score itself.
+    """
+
+    def __init__(
+        self,
+        output_key: str = FAITHFULNESS,
+    ):
+        """FaithfulnessScore initializer.
+
+        :param output_key: The key corresponding to the faithfulness score that
+            will be added to the input record.
+        """
+        super().__init__(output_key)
+        self.register_input_output_keys(
+            input_keys=[],
+            output_keys=[output_key],
+        )
+        self.output_key = output_key
+
+    @validate_call
+    def __call__(self, record: Dict[str, Any]) -> Dict[str, Any]:
+        """Augment the input record with the computed faithfulness score.
+
+        :param record: The input record.
+        :returns: The input record, with the faithfulness score added in.
+        """
+        verdict_output = record["raw_verdicts"]
+        statements = record["statements"]
+        record[self.output_key] = self._get_score(verdict_output, statements)
+        return record
+
+    @staticmethod
+    def _get_score(verdict_output: str, statements: str) -> float:
+        """Given generated statements and verdicts, compute Faithfulness score.
+
+        :param verdict_output: Verdicts(Yes/No) and explanations string get from Judge model.
+        :param statements: Statements string get from `GetStatements` Transform.
+        :returns: 0 to 1. See the docstring for `Faithfulness` for more details
+            on what these numerical values represent.
+        """
+        output = verdict_output.lower().strip()
+        num_statements = len(statements.split("\n"))
+        # TODO: handle edge case that num_statements is 0?
+
+        # calculate score
+        final_answer = "Final verdicts in order:"
```

Comment:
If we ask only one statement, why are we asking it in order?

Reply:
I think this could be another source of judge model hallucination. I think we can directly parse the individual verdicts from each statement rather than from `Final verdict for each statement in order:` string. How can we be sure that the parsed verdicts from `Final verdict for each statement in order:` are faithful to the verdicts mentioned in each individual statement.